net/raft/raft.go

```diff
@@ -734,17 +738,17 @@ func (sv *Service) join(addr, baseURL string) (id uint64, walobj *wal.WAL, err e
 	}
 
 	if !raft.IsEmptySnap(raftSnap) {
-		err := sv.saveSnapshot(&raftSnap)
-		if err != nil {
-			return 0, nil, errors.Wrap(err)
-		}
 		err = walobj.SaveSnapshot(walpb.Snapshot{
 			Index: raftSnap.Metadata.Index,
 			Term:  raftSnap.Metadata.Term,
 		})
 		if err != nil {
 			return 0, nil, errors.Wrap(err)
 		}
+		err := sv.saveSnapshot(&raftSnap)
```

Comment:
I wonder if `sv.saveSnapshot` and `wal.SaveSnapshot` could be consolidated somehow, to prevent us from accidentally doing these in the wrong order. (They always get executed together, right?)